velox/exec/Exchange.cpp

```diff
@@ -58,127 +60,110 @@ void SerializedPage::prepareStreamForDeserialize(ByteStream* input) {
 std::shared_ptr<ExchangeSource> ExchangeSource::create(
     const std::string& taskId,
     int destination,
-    std::shared_ptr<ExchangeQueue> queue) {
+    std::shared_ptr<ExchangeQueue> queue,
+    memory::MemoryPool* pool) {
   for (auto& factory : factories()) {
-    auto result = factory(taskId, destination, queue);
+    auto result = factory(taskId, destination, queue, pool);
     if (result) {
       return result;
     }
   }
   VELOX_FAIL("No ExchangeSource factory matches {}", taskId);
 }
 
-void ExchangeSource::setMemoryPool(memory::MemoryPool* pool) {
-  pool_ = pool;
-}
-
 // static
 std::vector<ExchangeSource::Factory>& ExchangeSource::factories() {
   static std::vector<Factory> factories;
   return factories;
 }
 
-namespace {
-class LocalExchangeSource : public ExchangeSource {
```

Comment:
Exposing LocalExchangeSource to build a memory pool supported LocalExchangeSource for testing.

Reply:
This is no longer needed.